system/settings/settings.xml

```diff
@@ -1080,6 +1080,19 @@
           </dependencies>
           <control type="toggle" />
         </setting>
+        <setting id="filelists.confirmfiledelete" type="boolean" label="40801" help="40802">
+          <level>2</level>
+          <default>false</default>
+          <dependencies>
+            <dependency type="enable">
+              <or>
+                <condition on="property" operator="!is" name="ProfileHasFilesLocked" />
+                <condition on="property" name="IsMasterUser" />
+              </or>
+            </dependency>
```

Comment:
Can you explain why we need this dependency? Just asking.

Reply:
Same as the one on Allow file deletion. Revise if needed.

Reply:
Then this rather should be a dependency to "allow file deletion". Enable this setting only if "allow file deletion" is enabled.